app/daemon.go

```diff
@@ -633,7 +633,7 @@ func (d *MenderShellDaemon) routeMessageStopShell(message *ws.ProtoMsg) error {
 			}
 		}
 		d.routeMessageResponse(response, err)
-		return err
+		return session.MenderShellDeleteById(s.GetId())
```

Comment:
I don't think this is correct, we need to return `err` regardless of `session.MenderShellDeleteById(s.GetId())` succession or not.

If any, we could wrap `err` with the potential error of `session.MenderShellDeleteById(s.GetId())`